src/lib/pipe_lib/strict_pipe.ml

```diff
@@ -84,6 +84,28 @@ module Reader = struct
 
     let iter_sync readers ~f = iter readers ~f:(fun x -> f x ; Deferred.unit)
   end
+
+  module Fork = struct
+    let n reader count =
+      let pipes = List.init count ~f:(fun _ -> Pipe.create ()) in
+      let writers = List.map pipes ~f:(fun (_, w) -> w) in
+      let readers = List.map pipes ~f:(fun (r, _) -> r) in
+      don't_wait_for
+        (iter reader ~f:(fun x ->
+             Deferred.List.iter writers ~f:(fun writer ->
+                 if not (Pipe.is_closed writer) then Pipe.write writer x
+                 else return () ) )) ;
+      don't_wait_for
+        (let%map () =
+           Deferred.List.iter readers
+             ~f:(Fn.compose Deferred.return Pipe.close_read)
```

Comment:
This would immediately close all of the pipes, right? What's even the point of this deferred if it does nothing? (The argument to `Deferred.return` is evaluated immediately).

Reply:
yes this should be `Pipe.closed`, fixing